modules/extensions-support/src/main/java/org/mule/module/extension/internal/runtime/resolver/ConnectorArgumentResolver.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.runtime.resolver;
+
+import static org.mule.util.Preconditions.checkArgument;
+import org.mule.extension.runtime.OperationContext;
+import org.mule.module.extension.internal.ExtensionProperties;
+
+/**
+ * An implementation of {@link ArgumentResolver} which returns the value of
+ * the {@link ExtensionProperties#CONNECTION_PARAM} parameter, which is expected
+ * to have been previously set on the supplied {@link OperationContext}
+ * <p/>
+ * Because this {@link ArgumentResolver} is stateless and thread-safe,
```

Comment:
To me these conditions does not mean that the class must be a singleton. If there is only one instance used on MethodArgumentResolverDelegate, then that class should maintain the instance instead of making this class a singleton.